src/js/utils/index.d.ts

```diff
@@ -24,6 +24,7 @@ export interface DeepMerge {
   <T extends object, S extends object[]>(target: T, ...sources: S): T & S[number];
 }
 
+export type MarginType = "none" | "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | {bottom?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,horizontal?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,left?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,right?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,top?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string,vertical?: "xxsmall" | "xsmall" | "small" | "medium" | "large" | "xlarge" | string} | string;
```

Comment:
Since we'll be pursuing this approach for more props, I'm thinking we should have a section on this file that is separated from the functions and the other definitions, and it will be dedicated just for props. 
At the end of the file, we could have a comment that is explaining that, and the MarginType as the first prop.

Reply:
I see you added it on the bottom, can we remove the duplication?